[BUG] OpenSearch source plugin requires auth configuration even when the source cluster does not need any

**Describe the bug**
Firstly, thanks for merging in [support for the scroll API for the ElasticSearch accessor](https://github.com/opensearch-project/data-prepper/pull/2930)! When I attempted to use this plugin with a test ElasticSearch OSS source cluster, I ran into the following error:

```
....
2023-06-26T22:30:19,634 [historical-data-migration-sink-worker-2-thread-1] INFO  org.opensearch.dataprepper.pipeline.Pipeline - Pipeline [historical-data-migration] Sink is ready, starting source...
2023-06-26T22:30:19,641 [historical-data-migration-sink-worker-2-thread-1] INFO  org.opensearch.dataprepper.plugins.source.opensearch.worker.client.OpenSearchClientFactory - Using username and password for auth for the OpenSearch source
2023-06-26T22:30:19,642 [historical-data-migration-sink-worker-2-thread-1] ERROR org.opensearch.dataprepper.pipeline.common.PipelineThreadPoolExecutor - Pipeline [historical-data-migration] process worker encountered a fatal exception, cannot proceed further
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Username may not be null
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
	at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
	at org.opensearch.dataprepper.pipeline.common.PipelineThreadPoolExecutor.afterExecute(PipelineThreadPoolExecutor.java:70) [data-prepper-core-2.4.0-SNAPSHOT.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.IllegalArgumentException: Username may not be null
	at org.apache.http.util.Args.notNull(Args.java:54) ~[httpcore-4.4.16.jar:4.4.16]
	at org.apache.http.auth.UsernamePasswordCredentials.<init>(UsernamePasswordCredentials.java:81) ~[httpclient-4.5.14.jar:4.5.14]
	at org.opensearch.dataprepper.plugins.source.opensearch.worker.client.OpenSearchClientFactory.attachUsernamePassword(OpenSearchClientFactory.java:175) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.source.opensearch.worker.client.OpenSearchClientFactory.createOpenSearchRestClient(OpenSearchClientFactory.java:137) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.source.opensearch.worker.client.OpenSearchClientFactory.provideOpenSearchClient(OpenSearchClientFactory.java:75) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.source.opensearch.worker.client.SearchAccessorStrategy.getSearchAccessor(SearchAccessorStrategy.java:61) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSource.startProcess(OpenSearchSource.java:49) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSource.start(OpenSearchSource.java:41) ~[opensearch-source-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.pipeline.Pipeline.startSourceAndProcessors(Pipeline.java:210) ~[data-prepper-core-2.4.0-SNAPSHOT.jar:?]
	at org.opensearch.dataprepper.pipeline.Pipeline.lambda$execute$2(Pipeline.java:251) ~[data-prepper-core-2.4.0-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	... 2 more
2023-06-26T22:30:19,654 [historical-data-migration-sink-worker-2-thread-1] INFO  org.opensearch.dataprepper.pipeline.Pipeline - Pipeline [historical-data-migration] - Received shutdown signal with processor shutdown timeout PT30S and sink shutdown timeout PT30S. Initiating the shutdown process
...
```

It looks like the plugin is expecting `username` and `password` configurations in the pipeline even when the source cluster does not have security enabled (which mine does not since i'm using the OSS distro and security is an X-Pack plugin)

**To Reproduce**
Steps to reproduce the behavior:
1. Start/use an ElasticSearch OSS source cluster (i'm using [ES OSS 7.10.2](https://www.elastic.co/downloads/past-releases/elasticsearch-oss-7-10-2)) without any auth
2. Configure a Data Prepper pipeline with the following source:
```
test-pipeline:
    source:
        opensearch:
            hosts: ["<es-oss-host>"]
```
3. Start Data Prepper using this pipeline and observe the error

**Expected behavior**
The above source configuration should work without errors

**Screenshots**
N/A, stack-trace provided above

**Environment (please complete the following information):**
 - OS: MacOS 12.6.6
 - Version: Data Prepper 2.4.0-SNAPSHOT

**Additional context**
I am able to work around this problem by providing dummy values for username and password in the config:

```
test-pipeline:
    source:
        opensearch:
            hosts: ["<es-oss-host>"]
            username: "dummy"
            password: "dummy"
```

 The ES source cluster ignores/does not parse these values so the API call goes through.

--- a/data-prepper-plugins/opensearch-source/README.md
+++ b/data-prepper-plugins/opensearch-source/README.md
@@ -114,6 +114,9 @@ opensearch-source-pipeline:
 - `password` (Optional) : A String of password used in the internal users of OpenSearch cluster. Default is null.
 
 
+- `disable_authentication` (Optional) : A boolean that can disable authentication if the cluster supports it. Defaults to false.
+
+
 - `aws` (Optional) : AWS configurations. See [AWS Configuration](#aws_configuration) for details. SigV4 is enabled by default when this option is used.
 
 

--- a/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSource.java
+++ b/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSource.java
@@ -31,6 +31,8 @@ public class OpenSearchSource implements Source<Record<Event>>, UsesSourceCoordi
                             final AwsCredentialsSupplier awsCredentialsSupplier) {
         this.openSearchSourceConfiguration = openSearchSourceConfiguration;
         this.awsCredentialsSupplier = awsCredentialsSupplier;
+
+        openSearchSourceConfiguration.validateAwsConfigWithUsernameAndPassword();
     }
 
     @Override

--- a/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfiguration.java
+++ b/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfiguration.java
@@ -6,9 +6,8 @@ package org.opensearch.dataprepper.plugins.source.opensearch;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import jakarta.validation.Valid;
-import jakarta.validation.constraints.AssertTrue;
-import jakarta.validation.constraints.Min;
 import jakarta.validation.constraints.NotNull;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
 import org.opensearch.dataprepper.plugins.source.opensearch.configuration.AwsAuthenticationConfiguration;
 import org.opensearch.dataprepper.plugins.source.opensearch.configuration.ConnectionConfiguration;
 import org.opensearch.dataprepper.plugins.source.opensearch.configuration.IndexParametersConfiguration;
@@ -20,13 +19,6 @@ import java.util.Objects;
 
 public class OpenSearchSourceConfiguration {
 
-    /**
-     * 0 indicates infinite retries
-     */
-    @JsonProperty("max_retries")
-    @Min(0)
-    private Integer maxRetries = 0;
-
     @NotNull
     @JsonProperty("hosts")
     private List<String> hosts;
@@ -36,6 +28,9 @@ public class OpenSearchSourceConfiguration {
 
     @JsonProperty("password")
     private String password;
+
+    @JsonProperty("disable_authentication")
+    private Boolean disableAuthentication = false;
 
     @JsonProperty("connection")
     @Valid
@@ -57,10 +52,6 @@ public class OpenSearchSourceConfiguration {
     @Valid
     private SearchConfiguration searchConfiguration = new SearchConfiguration();
 
-    public Integer getMaxRetries() {
-        return maxRetries;
-    }
-
     public List<String> getHosts() {
         return hosts;
     }
@@ -72,6 +63,8 @@ public class OpenSearchSourceConfiguration {
     public String getPassword() {
         return password;
     }
+
+    public Boolean isAuthenticationDisabled() { return disableAuthentication; }
 
     public ConnectionConfiguration getConnectionConfiguration() {
         return connectionConfiguration;
@@ -93,10 +86,13 @@ public class OpenSearchSourceConfiguration {
         return searchConfiguration;
     }
 
-    @AssertTrue(message = "Either username and password, or aws options must be specified. Both cannot be set at once.")
-    boolean validateAwsConfigWithUsernameAndPassword() {
-        return !((Objects.nonNull(awsAuthenticationOptions) && (Objects.nonNull(username) || Objects.nonNull(password))) ||
-                (Objects.isNull(awsAuthenticationOptions) && (Objects.isNull(username) || Objects.isNull(password))));
+    void validateAwsConfigWithUsernameAndPassword() {
+
+        if (((Objects.nonNull(awsAuthenticationOptions) && ((Objects.nonNull(username) || Objects.nonNull(password)) || disableAuthentication)) ||
+                (Objects.nonNull(username) || Objects.nonNull(password)) && disableAuthentication) ||
+                (Objects.isNull(awsAuthenticationOptions) && (Objects.isNull(username) || Objects.isNull(password)) && !disableAuthentication)) {
+            throw new InvalidPluginConfigurationException("Either username and password, or aws options must be specified. Both cannot be set at once. Authentication can be disabled by setting the disable_authentication flag to true.");
+        }
     }
 
 }

--- a/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactory.java
+++ b/data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactory.java
@@ -133,8 +133,7 @@ public class OpenSearchClientFactory {
 
         final RestClientBuilder restClientBuilder = RestClient.builder(httpHosts);
 
-        LOG.info("Using username and password for auth for the OpenSearch source");
-        attachUsernamePassword(restClientBuilder, openSearchSourceConfiguration);
+        attachBasicAuth(restClientBuilder, openSearchSourceConfiguration);
 
         setConnectAndSocketTimeout(restClientBuilder, openSearchSourceConfiguration);
 
@@ -161,33 +160,36 @@ public class OpenSearchClientFactory {
                 new BasicHeader("Content-type", "application/json")
         });
 
-        LOG.info("Using username and password for auth for the OpenSearch source");
-        attachUsernamePassword(restClientBuilder, openSearchSourceConfiguration);
-
+        attachBasicAuth(restClientBuilder, openSearchSourceConfiguration);
         setConnectAndSocketTimeout(restClientBuilder, openSearchSourceConfiguration);
 
         return restClientBuilder.build();
     }
 
-    private void attachUsernamePassword(final RestClientBuilder restClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
-        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-        credentialsProvider.setCredentials(AuthScope.ANY,
-                new UsernamePasswordCredentials(openSearchSourceConfiguration.getUsername(), openSearchSourceConfiguration.getPassword()));
+    private void attachBasicAuth(final RestClientBuilder restClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
 
         restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> {
-            httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+            if (!openSearchSourceConfiguration.isAuthenticationDisabled()) {
+                attachUsernameAndPassword(httpClientBuilder, openSearchSourceConfiguration);
+            } else {
+                LOG.warn("Authentication was explicitly disabled for the OpenSearch source");
+            }
+
             attachSSLContext(httpClientBuilder, openSearchSourceConfiguration);
             return httpClientBuilder;
         });
     }
 
-    private void attachUsernamePassword(final org.elasticsearch.client.RestClientBuilder restClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
-        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-        credentialsProvider.setCredentials(AuthScope.ANY,
-                new UsernamePasswordCredentials(openSearchSourceConfiguration.getUsername(), openSearchSourceConfiguration.getPassword()));
+    private void attachBasicAuth(final org.elasticsearch.client.RestClientBuilder restClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
 
         restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> {
-            httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+
+            if (!openSearchSourceConfiguration.isAuthenticationDisabled()) {
+                attachUsernameAndPassword(httpClientBuilder, openSearchSourceConfiguration);
+            } else {
+                LOG.warn("Authentication was explicitly disabled for the OpenSearch source");
+            }
+
             attachSSLContext(httpClientBuilder, openSearchSourceConfiguration);
             httpClientBuilder.addInterceptorLast(
                     (HttpResponseInterceptor)
@@ -209,6 +211,15 @@ public class OpenSearchClientFactory {
 
             return requestConfigBuilder;
         });
+    }
+
+    private void attachUsernameAndPassword(final HttpAsyncClientBuilder httpClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
+        LOG.info("Using username and password for auth for the OpenSearch source");
+
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY,
+                new UsernamePasswordCredentials(openSearchSourceConfiguration.getUsername(), openSearchSourceConfiguration.getPassword()));
+        httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
     }
 
     private void setConnectAndSocketTimeout(final org.elasticsearch.client.RestClientBuilder restClientBuilder, final OpenSearchSourceConfiguration openSearchSourceConfiguration) {

--- a/data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfigurationTest.java
+++ b/data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfigurationTest.java
@@ -9,10 +9,12 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
 import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class OpenSearchSourceConfigurationTest {
 
@@ -21,7 +23,7 @@ public class OpenSearchSourceConfigurationTest {
     @Test
     void open_search_source_username_password_only() throws JsonProcessingException {
 
-        final String sourceConfigurationYaml = "max_retries: 5\n" +
+        final String sourceConfigurationYaml =
                 "hosts: [\"http://localhost:9200\"]\n" +
                 "username: test\n" +
                 "password: test\n" +
@@ -44,18 +46,49 @@ public class OpenSearchSourceConfigurationTest {
         assertThat(sourceConfiguration.getIndexParametersConfiguration(), notNullValue());
         assertThat(sourceConfiguration.getSchedulingParameterConfiguration(), notNullValue());
         assertThat(sourceConfiguration.getHosts(), notNullValue());
-        assertThat(sourceConfiguration.getMaxRetries(), equalTo(5));
 
-        assertThat(sourceConfiguration.validateAwsConfigWithUsernameAndPassword(), equalTo(true));
+        sourceConfiguration.validateAwsConfigWithUsernameAndPassword();
         assertThat(sourceConfiguration.getPassword(), equalTo("test"));
         assertThat(sourceConfiguration.getUsername(), equalTo("test"));
         assertThat(sourceConfiguration.getAwsAuthenticationOptions(), equalTo(null));
     }
 
     @Test
-    void opensearch_source_aws_only() throws JsonProcessingException {
-        final String sourceConfigurationYaml = "max_retries: 5\n" +
+    void open_search_disabled_authentication() throws JsonProcessingException {
+
+        final String sourceConfigurationYaml =
                 "hosts: [\"http://localhost:9200\"]\n" +
+                        "disable_authentication: true\n" +
+                        "connection:\n" +
+                        "  insecure: true\n" +
+                        "  cert: \"cert\"\n" +
+                        "indices:\n" +
+                        "  include:\n" +
+                        "    - index_name_regex: \"regex\"\n" +
+                        "    - index_name_regex: \"regex-two\"\n" +
+                        "scheduling:\n" +
+                        "  job_count: 3\n" +
+                        "search_options:\n" +
+                        "  batch_size: 1000\n" +
+                        "  query: \"test\"\n";
+        final OpenSearchSourceConfiguration sourceConfiguration = objectMapper.readValue(sourceConfigurationYaml, OpenSearchSourceConfiguration.class);
+
+        assertThat(sourceConfiguration.getSearchConfiguration(), notNullValue());
+        assertThat(sourceConfiguration.getConnectionConfiguration(), notNullValue());
+        assertThat(sourceConfiguration.getIndexParametersConfiguration(), notNullValue());
+        assertThat(sourceConfiguration.getSchedulingParameterConfiguration(), notNullValue());
+        assertThat(sourceConfiguration.getHosts(), notNullValue());
+
+        sourceConfiguration.validateAwsConfigWithUsernameAndPassword();
+        assertThat(sourceConfiguration.isAuthenticationDisabled(), equalTo(true));
+        assertThat(sourceConfiguration.getPassword(), equalTo(null));
+        assertThat(sourceConfiguration.getUsername(), equalTo(null));
+        assertThat(sourceConfiguration.getAwsAuthenticationOptions(), equalTo(null));
+    }
+
+    @Test
+    void opensearch_source_aws_only() throws JsonProcessingException {
+        final String sourceConfigurationYaml = "hosts: [\"http://localhost:9200\"]\n" +
                 "connection:\n" +
                 "  insecure: true\n" +
                 "  cert: \"cert\"\n" +
@@ -74,7 +107,7 @@ public class OpenSearchSourceConfigurationTest {
 
         final OpenSearchSourceConfiguration sourceConfiguration = objectMapper.readValue(sourceConfigurationYaml, OpenSearchSourceConfiguration.class);
 
-        assertThat(sourceConfiguration.validateAwsConfigWithUsernameAndPassword(), equalTo(true));
+        sourceConfiguration.validateAwsConfigWithUsernameAndPassword();
         assertThat(sourceConfiguration.getPassword(), equalTo(null));
         assertThat(sourceConfiguration.getUsername(), equalTo(null));
         assertThat(sourceConfiguration.getAwsAuthenticationOptions(), notNullValue());
@@ -85,8 +118,7 @@ public class OpenSearchSourceConfigurationTest {
 
     @Test
     void opensearch_source_aws_sts_external_id() throws JsonProcessingException {
-        final String sourceConfigurationYaml = "max_retries: 5\n" +
-            "hosts: [\"http://localhost:9200\"]\n" +
+        final String sourceConfigurationYaml = "hosts: [\"http://localhost:9200\"]\n" +
             "connection:\n" +
             "  insecure: true\n" +
             "  cert: \"cert\"\n" +
@@ -106,7 +138,7 @@ public class OpenSearchSourceConfigurationTest {
 
         final OpenSearchSourceConfiguration sourceConfiguration = objectMapper.readValue(sourceConfigurationYaml, OpenSearchSourceConfiguration.class);
 
-        assertThat(sourceConfiguration.validateAwsConfigWithUsernameAndPassword(), equalTo(true));
+        sourceConfiguration.validateAwsConfigWithUsernameAndPassword();
         assertThat(sourceConfiguration.getPassword(), equalTo(null));
         assertThat(sourceConfiguration.getUsername(), equalTo(null));
         assertThat(sourceConfiguration.getAwsAuthenticationOptions(), notNullValue());
@@ -141,14 +173,12 @@ public class OpenSearchSourceConfigurationTest {
 
         final OpenSearchSourceConfiguration sourceConfiguration = objectMapper.readValue(sourceConfigurationYaml, OpenSearchSourceConfiguration.class);
 
-        assertThat(sourceConfiguration.validateAwsConfigWithUsernameAndPassword(), equalTo(false));
-        assertThat(sourceConfiguration.getMaxRetries(), equalTo(0));
+        assertThrows(InvalidPluginConfigurationException.class, sourceConfiguration::validateAwsConfigWithUsernameAndPassword);
     }
 
     @Test
-    void one_of_username_password_or_aws_config_is_required() throws JsonProcessingException {
+    void one_of_username_password_or_aws_config_or_authDisabled_is_required() throws JsonProcessingException {
         final String sourceConfigurationYaml =
-                "max_retries: 5\n" +
                         "hosts: [\"http://localhost:9200\"]\n" +
                         "connection:\n" +
                         "  insecure: true\n" +
@@ -165,6 +195,6 @@ public class OpenSearchSourceConfigurationTest {
 
         final OpenSearchSourceConfiguration sourceConfiguration = objectMapper.readValue(sourceConfigurationYaml, OpenSearchSourceConfiguration.class);
 
-        assertThat(sourceConfiguration.validateAwsConfigWithUsernameAndPassword(), equalTo(false));
+        assertThrows(InvalidPluginConfigurationException.class, sourceConfiguration::validateAwsConfigWithUsernameAndPassword);
     }
 }

--- a/data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java
+++ b/data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactoryTest.java
@@ -29,6 +29,8 @@ import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoInteractions;
 import static org.mockito.Mockito.when;
 
@@ -118,5 +120,39 @@ public class OpenSearchClientFactoryTest {
         assertThat(awsCredentialsOptions.getRegion(), equalTo(Region.US_EAST_1));
         assertThat(awsCredentialsOptions.getStsHeaderOverrides(), equalTo(Collections.emptyMap()));
         assertThat(awsCredentialsOptions.getStsRoleArn(), equalTo(stsRoleArn));
+    }
+
+    @Test
+    void provideElasticSearchClient_with_auth_disabled() {
+        when(openSearchSourceConfiguration.isAuthenticationDisabled()).thenReturn(true);
+
+        when(connectionConfiguration.getCertPath()).thenReturn(null);
+        when(connectionConfiguration.getSocketTimeout()).thenReturn(null);
+        when(connectionConfiguration.getConnectTimeout()).thenReturn(null);
+        when(connectionConfiguration.isInsecure()).thenReturn(true);
+
+        final ElasticsearchClient elasticsearchClient = createObjectUnderTest().provideElasticSearchClient(openSearchSourceConfiguration);
+        assertThat(elasticsearchClient, notNullValue());
+
+        verifyNoInteractions(awsCredentialsSupplier);
+        verify(openSearchSourceConfiguration, never()).getUsername();
+        verify(openSearchSourceConfiguration, never()).getPassword();
+    }
+
+    @Test
+    void provideOpenSearchClient_with_auth_disabled() {
+        when(openSearchSourceConfiguration.isAuthenticationDisabled()).thenReturn(true);
+
+        when(connectionConfiguration.getCertPath()).thenReturn(null);
+        when(connectionConfiguration.getSocketTimeout()).thenReturn(null);
+        when(connectionConfiguration.getConnectTimeout()).thenReturn(null);
+        when(connectionConfiguration.isInsecure()).thenReturn(true);
+
+        final OpenSearchClient openSearchClient = createObjectUnderTest().provideOpenSearchClient(openSearchSourceConfiguration);
+        assertThat(openSearchClient, notNullValue());
+
+        verifyNoInteractions(awsCredentialsSupplier);
+        verify(openSearchSourceConfiguration, never()).getUsername();
+        verify(openSearchSourceConfiguration, never()).getPassword();
     }
 }

